modules/core/task/include/task.hpp

```diff
@@ -10,45 +10,41 @@
 
 namespace ppc::core {
 
-struct TaskData {
-  std::vector<uint8_t *> inputs;
-  std::vector<std::uint32_t> inputs_count;
-  std::vector<uint8_t *> outputs;
-  std::vector<std::uint32_t> outputs_count;
-  enum StateOfTesting { FUNC, PERF } state_of_testing;
+template <typename InType, typename OutType>
+struct GenericTaskData {
+  const std::span<InType> input;
+  std::span<OutType> output;
+
+  GenericTaskData(const std::span<InType>& input_, std::span<OutType> output_)
+      : input(input_), output(std::move(output_)) {}
+  GenericTaskData(const InType& input_, OutType& output_)
+      : input(std::addressof(input_), 1), output(std::addressof(output_), 1) {}
+  GenericTaskData(const std::span<InType>& input_, OutType& output_)
+      : input(input_), output(std::addressof(output_), 1) {}
+  GenericTaskData(const InType& input_, std::span<OutType> output_)
+      : input(std::addressof(input_), 1), output(std::move(output_)) {}
 };
```

Comment:
please write code snippet, how yours GenericTaskData describe input - vector(uint8), vector(float), and output vector(uint8) and 1 element float?

Reply:
I'd say it depends on the task: it's possible to use `std::pair<std::vector<uint8_t>, std::vector<float>>` or `std::tuple`. If it's a more complex structure, I'd create a tiny struct descriptor for it.